File: trunk/nexml/javascript/Phylo/Taxa.js
var constant = new Constant();
var mediator = new TaxaMediator();

function Taxa(args) {
	if (args==null) args = {};
	args["tag"] = "otus";	
    this.Listable(args);
    this._type      = constant._TAXA_;
    this._container = constant._NONE_;
    return this;
}
copyPrototype(Taxa,Listable);

Taxa.prototype.set_forest = function (forest) {
    if ( looks_like_object(forest,constant._FOREST_) ) {
        forest.set_taxa(this);
    }
    return this;
};

Taxa.prototype.set_matrix = function (matrix) {
    if ( looks_like_object(matrix,constant._MATRIX_) ) {
        matrix.set_taxa(this);
    }
    return this;
};

Taxa.prototype.unset_forest = function(forest) {
    if ( looks_like_object(forest,constant._FOREST_) ) {
        forest.unset_taxa();
    }
    return this;
};

Taxa.prototype.unset_matrix = function(matrix) {
    if ( looks_like_object(matrix,constant._MATRIX_) ) {
        matrix.unset_taxa();
    }
    return this;
};

Taxa.prototype.get_forests = function() {
    return mediator.get_link( {
        "source" : this,
        "type"   : constant._FOREST_()
    } );
};

Taxa.prototype.get_matrices = function () {
    return mediator.get_link( {
        "source" : this,
        "type"   : constant._MATRIX_()    
    } );
};

Taxa.prototype.get_ntax = function () {
    return this.get_entities().length;
};

Taxa.prototype._type = function () {
    return this._type;
};

Taxa.prototype._container = function() {
    return this._container;
};

File: nexml/java/src/org/biophylo/Taxa/Taxa.java
package org.biophylo.Taxa;
import org.biophylo.Mediators.TaxaMediator;
import org.biophylo.Util.*;
import org.biophylo.Listable;
import org.biophylo.Forest.*;
import java.util.*;

public class Taxa extends Listable {
	private static TaxaMediator taxaMediator = TaxaMediator.getInstance();
	private static Logger logger = Logger.getInstance();
	
	/**
	 * 
	 */
	public Taxa () {
		super();
		this.type = CONSTANT.TAXA;
		this.container = CONSTANT.PROJECT;
		this.tag = "otus";
	}
	
	/**
	 * @param forest
	 */
	public void setForest(Forest forest) {
		forest.setTaxa(this);
	}
	
	/**
	 * @param forest
	 */
	public void unsetForest(Forest forest) {
		forest.unsetTaxa();
	}
	
	/**
	 * @return
	 */
	public Forest[] getForests () {
		Vector tl = taxaMediator.getLink(this.getId(), CONSTANT.FOREST);
		for ( int i = 0; i < tl.size(); i++ ) {
			logger.debug(""+tl.get(i));
		}
		Forest[] f = new Forest[tl.size()];
		tl.copyInto(f);
		return f;
	}
	
	/**
	 * @return
	 */
	public int getNtax() {
		return this.getEntities().length;
	}
	
}
